vinyldns/resource_zone_test.go

```diff
@@ -92,35 +100,49 @@ func testAccVinylDNSZoneDestroy(s *terraform.State) error {
 
 func testAccCheckVinylDNSZoneExists(n string) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
-		rs, ok := s.RootModule().Resources[n]
-		if !ok {
-			return fmt.Errorf("Not found %s", rs)
-		}
-		log.Printf("[INFO] testing that zone exists: %s", rs.Primary.ID)
+		return testAccCheckZoneExists("email@foo.com", n, s)
+	}
+}
 
-		if rs.Primary.ID == "" {
-			return fmt.Errorf("No Zone ID is set")
-		}
+func testAccCheckVinylDNSZoneUpdatedExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		return testAccCheckZoneExists("updated_email@foo.com", n, s)
+	}
+}
 
-		client := testAccProvider.Meta().(*vinyldns.Client)
+func testAccCheckZoneExists(email string, n string, s *terraform.State) error {
+	rs, ok := s.RootModule().Resources[n]
+	if !ok {
+		return fmt.Errorf("Not found %s", rs)
+	}
+	log.Printf("[INFO] testing that zone exists: %s", rs.Primary.ID)
 
-		readZone, err := client.Zone(rs.Primary.ID)
-		if err != nil {
-			return err
-		}
+	if rs.Primary.ID == "" {
+		return fmt.Errorf("No Zone ID is set")
+	}
 
-		if readZone.Name != "system-test." {
-			return fmt.Errorf("Zone not found")
-		}
+	client := testAccProvider.Meta().(*vinyldns.Client)
+
+	readZone, err := client.Zone(rs.Primary.ID)
+	if err != nil {
+		return err
+	}
+
+	if readZone.Name != "system-test." {
+		return fmt.Errorf("Zone %s not found", "system-test.")
+	}
 
-		return nil
+	if readZone.Email != email {
+		return fmt.Errorf("Zone %s with email %s not found", "system-test.", email)
 	}
+
+	return nil
 }
 
-const testAccVinylDNSZoneConfigBasic = `
+func testAccVinylDNSZoneConfigBasic(email string) string {
+	return fmt.Sprintf(`
```

Comment:
Small nitpick: Multiline string literals as function arguments makes this code a bit awkward to read. Declaring a `const` and passing that instead makes it a bit clearer IMO.

```
func f(email string) string {
  const t = `
resource "example provider" "example resource" {
  email = "%s"
}
`

  return fmt.Sprintf(t, email)
}
```